lib/src/main/kotlin/BST.kt

```diff
@@ -0,0 +1,163 @@
+package BSTree
+
+import TreeMap.*
+
+
+/**
+ * Node from a binary search tree
+ */
+class BSTNode<K : Comparable<K>, V>(key: K, value: V, parent: BSTNode<K, V>?) : Node<K, V, BSTNode<K, V>>(key, value, parent)
```

Comment:
Nit: кажется, поставить для parent null значением по умолчанию звучит как хорошая идея
```suggestion
class BSTNode<K : Comparable<K>, V>(key: K, value: V, parent: BSTNode<K, V>?) : Node<K, V, BSTNode<K, V>>(key, value, parent = null)
```